lib/coverband/integrations/resque.rb

```diff
@@ -2,9 +2,11 @@
 
 Resque.after_fork do |job|
   Coverband.start
+  Coverband.runtime_coverage!
 end
 
 Resque.before_first_fork do
+  Coverband.eager_loading_coverage!
```

Comment:
Interesting, I had thought the railtie would handle this if we are running resque within rails.

Reply:
I am fixing the test, but resque workers don't seem to call the full rails initialization, I will dig a bit more into why the rails hooks aren't taking care of this... but in our setup without this change while it does seem to preload our app code it sends a first hit before running any job with coverage in the runtime state.

Reply:
also, I believe sidekiq is fine based on how it loads

Reply:
Oh. Good to know about sidekiq. Guess we’ll have to handle on a case by case basis.

Reply:
OK, so I believe the issue comes down to various ways to run resque but a common way is

`bundle exec rake environment resque:work`

This runs `rake environment` which boots the full rails app and then runs resque, but while our rails hooks fire, the files aren't loaded this is because of a Rails perf hack, `RAKE ignores eager_load`

```
# Rake tasks automatically ignore this option for performance.
config.eager_load = true
```
https://github.com/rails/rails/blob/df8ee09ce71338cdf9816225df1bdebc707f3560/actiontext/test/dummy/config/environments/production.rb#L12`

Resque in its task LATER forces eager_loading if the rails config option is set, but this happens AFTER our rails hooks have fired and set coverband back into runtime mode assuming that code loading is all done.

This took me far to long to figure out as did figuring out how to fix the resque test.

Reply:
Great work! Was looking into how they disable eager load and it looks like they just set it to false within the environment rake task.

https://github.com/rails/rails/commit/9cac69c602f57e397fe01d866cb24ce1781606d4#diff-580e183141be706162224ef696bdb6c7R294